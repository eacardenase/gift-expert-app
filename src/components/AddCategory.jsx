import { useState } from 'react';

export const AddCategory = ({ onAddCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        const newInputValue = inputValue.trim();

        if (newInputValue.length < 2) return;

        onAddCategory(newInputValue);
        setInputValue('');
    };

    return (
        <form onSubmit={handleSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    );
};
