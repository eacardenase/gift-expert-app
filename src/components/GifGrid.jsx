import PropTypes from 'prop-types';
import { useState, useEffect } from 'react';

import { getGifs } from '../helpers/getGifs';

const GifGrid = ({ category }) => {
    const [images, setImages] = useState([]);

    const getImages = async () => {
        const newImages = await getGifs(category);

        setImages(newImages);
    };

    useEffect(() => {
        getImages();
    }, []);

    return (
        <>
            <h3>{category}</h3>
            <ol>
                {images.map(({ url, id }) => {
                    return (
                        <li key={id}>
                            <img src={url} alt="gif" />
                        </li>
                    );
                })}
            </ol>
        </>
    );
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
};

export default GifGrid;
