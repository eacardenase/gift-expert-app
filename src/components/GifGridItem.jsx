import PropTypes from 'prop-types';

const GifGridItem = ({ url, title }) => {
    console.log(url);
    console.log(title);

    return (
        <div className="card">
            <img src={url} alt={title} />
            {title && <p>{title}</p>}
        </div>
    );
};

GifGridItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};

export default GifGridItem;
