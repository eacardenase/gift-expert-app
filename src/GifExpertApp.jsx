import { useState } from 'react';

import { AddCategory, GifGrid } from './components';

const GifExpertApp = () => {
    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = (newCategory) => {
        if (
            categories
                .map((category) => category.toLowerCase())
                .includes(newCategory.toLowerCase())
        )
            return;

        // setCategories((categories) => [ newCategory, ...categories]);
        setCategories([newCategory, ...categories]);
    };

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory onAddCategory={onAddCategory} />

            {categories.map((category) => (
                <GifGrid key={category} category={category} />
            ))}
        </>
    );
};

export default GifExpertApp;
