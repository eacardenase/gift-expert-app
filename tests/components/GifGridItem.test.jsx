import { describe, test, expect, beforeEach } from 'vitest';

import { cleanup, fireEvent, render, screen } from '@testing-library/react';

import { GifGridItem } from '../../src/components';

describe('<GifGridItem />', () => {
    beforeEach(cleanup);

    const title = 'Test title';
    const url = 'www.google.com';

    test('should match with the snapshoot', () => {
        const { container } = render(<GifGridItem url={url} title={title} />);

        expect(container).toMatchSnapshot();
    });

    test('should show the image with respective url and alt text', () => {
        render(<GifGridItem url={url} title={title} />);

        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('should show title on component', () => {
        render(<GifGridItem url={url} title={title} />);

        expect(screen.getByText(title)).toBeDefined();
    });
});
