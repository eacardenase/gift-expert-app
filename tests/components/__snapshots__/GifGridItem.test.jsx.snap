// Vitest Snapshot v1

exports[`<GifGridItem /> > should match with the snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="www.google.com"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
